src/metaswitch/clearwater/etcd_shared/plugin_utils.py

```diff
@@ -25,12 +25,11 @@ def run_command(command, namespace=None, log_error=True):
     passing to this function.
     """
     if namespace:
-        command = "ip netns exec {} ".format(namespace) + command
+        command = "ip netns exec {} ".format(namespace) + shlex.quote(command)
 
     # Pass the close_fds argument to avoid the pidfile lock being held by
     # child processes
-    p = subprocess.Popen(command,
-                         shell=True,
+    p = subprocess.Popen(shlex.split(command),
```

Comment:
What needs to be passed to `Popen` is `["ip", "netns", "exec", "signaling", "echo", "hello", "world"]`.  I'd probably push for this function taking the `command` parameter as `["echo", "hello", "world"]` but maybe it could take a string and `shlex.split()` it as the first thing it does.